plugins/uast-kotlin/src/org/jetbrains/uast/kotlin/declarations/KotlinUAnnotation.kt

```diff
@@ -64,7 +64,7 @@ abstract class KotlinUAnnotationBase<T : KtCallElement>(
     protected abstract fun computeClassDescriptor(): ClassDescriptor?
 
     override fun resolve(): PsiClass? = computeClassDescriptor()?.let {
-        sourcePsi.calleeExpression?.let { ktExpression -> resolveToDeclaration(ktExpression, it) }
+        sourcePsi.calleeExpression?.let { ktExpression -> resolveToDeclarationImpl(ktExpression, it) }
```

Comment:
Use `baseResolveProviderService.resolveToDeclaration` to be consistent with the other places?

Reply:
Not really. They're different, and quite FE1.0 specific in the sense that it requires corresponding descriptor: https://github.com/JetBrains/kotlin/blob/master/plugins/uast-kotlin/src/org/jetbrains/uast/kotlin/internal/kotlinInternalUastUtils.kt#L324 I intentionally didn't make it a part of base resolve provider service.